grype/pkg/purl_provider.go

```diff
@@ -193,8 +193,13 @@ func purlToPackage(rawLine string) (*Package, *pkg.Package, string, string, erro
 		version = fmt.Sprintf("%s:%s", epoch, purl.Version)
 	}
 
+	name := purl.Name
+	if pkgType == pkg.GoModulePkg {
```

Comment:
```suggestion
	if pkgType == pkg.GoModulePkg && purl.Namespace != "" {
```

Reply:
Updated to consider the purl.Namespace and added a test case